orb.yml

```diff
@@ -79,12 +79,12 @@ commands:
           - attach_workspace:
               at: << parameters.workspace >>
     - run: >
-        [[ -r setup.sh ]] && . setup.sh &&
+        [[ -r setup.sh ]] && . setup.sh ;
         groovy << parameters.sonatypedir >>/bin/NexusPublisher.groovy
         --username << parameters.username >>
         --password << parameters.password >>
         --serverurl << parameters.serverurl >>
-        --filename << parameters.workspace >>/<< parameters.filename >>
+        --filename $(if [ ! -z "<< parameters.workspace >>" ] ; then echo << parameters.workspace >>/ ; fi)<< parameters.filename >>
```

Comment:
I assume parameters.workspace gets substituted in from some sort of templating language, but that language doesn't have features to check about non-empty files at prep time, so we have to code that into the runtime ourselves?

I *do* think the echo would benefit from having its message wrapped in quotes for better readability.

Reply:
I suppose it could be tightened up slightly by losing the explicit if in favor of just an `&&`:
`$([ ! -z "<< parameters.workspace >>" && echo "<< parameters.workspace >>/")`

Reply:
and also invert the condition, but that might lose some nuance: `$([ -z "<< parameters.workspace >>" ] || echo "<< parameters.workspace >>/")`

Reply:
just reading now that `test -z` is for "zero-length strings". Now it's really disappointing that the template language couldn't do that.